### DISCUSSION TITLE
[Question]: Custom GUI for live streams across multiple Frigate endpoints

### DISCUSSION THREAD

**Discussion by jritsema** [blakeblackshear/frigate#11672] [Ask A Question] [answered]
### What is your question:

First off, I love this project!  Thank you to all who contribute.  I started small but have added cameras over time and have outgrown my Frigate server hardware.  I added an additional server and now have cameras split across two Frigate endpoints.  Using MQTT subscriptions, I built my own GUI that consolidates events across the Frigate endpoints (and combines other home device events) for playback.  

This is working well, however, I would like to add a single UI for live streaming.  I am a software engineer but not an expert in video/WebRTC, etc.  Can anyone recommend how I can leverage Frigate's API to build a custom UI that stream's many cameras across two different Frigate instances?  Thanks in advance.

**Comment by moandcompany** [comment]
Considering that you'e looking to do this for live streaming specifically,

Assuming a typical installation, you're likely using rtsp and go2rtc to connect to your various IPcams for each Frigate instance. Why not use off the shelf IPcam viewing software to either connect to those cameras directly, to their rtsp streams, o via go2tc and its restreams?

**Reply by jritsema** [reply]
That would be too easy! 😀 I'm a builder and would like to consume the live streams in my custom app. Has anyone done this with frigate? Or can point me to some docs or frigate code that consumes the streams?

**Reply by moandcompany** [reply]
> That would be too easy! 😀 I'm a builder and would like to consume the live streams in my custom app. Has anyone done this with frigate? Or can point me to some docs or frigate code that consumes the streams?

What are your expectations when describing doing this "with Frigate?" Frigate's last few major versions have gone toward consuming detect/record streams directly from an instance of `go2rtc` bundled with Frigate.

**Reply by jritsema** [reply]
Sorry, allow me to clarify.  What I'm after is the architecture of how the app consumes the live streams and displays them in GUI.  I'm not a React expert, but it appears that it's using a [custom WebRTC client component](https://github.com/blakeblackshear/frigate/tree/dev/web/src/components/player)?  I believe Frigate uses ffmpeg to consume RTSP streams directly from my cameras (gortc server).  Does Frigate run an RTC server or re-expose the stream somehow?  It seems like if I wanted to write my own web app that displays all my cam feeds, I will need to write some JS that talks uses WebRTC pointed at Frigate's server. Does that help?

**Reply by NickM-27** [reply, marked as answer]
Frigate uses go2rtc for this, the go2rtc has JS/html examples https://github.com/AlexxIT/go2rtc/tree/master/www
